corehq/apps/case_importer/views.py

```diff
@@ -257,3 +264,73 @@ def excel_commit(request, domain):
     request.session.pop(EXCEL_SESSION_ID, None)
 
     return HttpResponseRedirect(base.ImportCases.get_url(domain))
+
+
+def _get_bulk_case_upload_args_from_request(request, domain):
+    try:
+        upload_file = request.FILES["file"]
+        case_type = request.POST["case_type"]
+        search_field = request.POST['search_field']
+        create_new_cases = request.POST.get('create_new_cases') == 'on'
+        search_column = request.POST['search_column']
+        return upload_file, case_type, search_field, create_new_cases, search_column
+    except Exception:
+        raise ImporterError(
+                "Invalid post request."
+                "Submit the form with field 'file' and the required params")
+
+
+@csrf_exempt
+@require_POST
+@api_auth
+@require_can_edit_data
+def bulk_case_upload_api(request, domain, **kwargs):
+    try:
+        response = _bulk_case_upload_api(request, domain)
+        return response
+    except ImporterError as e:
+        error = get_importer_error_message(e)
+    except SpreadsheetFileExtError:
+        error = "Please upload file with extension .xls or .xlsx"
+        return json_response({'error_msg': _(error)})
+
+
+def _bulk_case_upload_api(request,domain):
+    upload_file, case_type, search_field, create_new_cases, search_column =_get_bulk_case_upload_args_from_request(request, domain)
```

Comment:
nit: that number of return values makes me uncomfortable. given that those are all one line pulls from the request args, this might be cleaner inline, without the helper function, plus it will make it slightly easier to figure out whats going on. this is a pretty weak preference and I could be convinced that its nicer this way.